pkg/zcache/local_cache.go

```diff
@@ -11,19 +11,31 @@ import (
 	"time"
 )
 
+const (
+	neverExpires = -1
+)
+
 type CacheItem struct {
 	Value     []byte `json:"value"`
 	ExpiresAt int64  `json:"expires_at"`
 }
 
 func NewCacheItem(value []byte, ttl time.Duration) CacheItem {
+	expiresAt := time.Now().Add(ttl).Unix()
```

Comment:
I think this is not going to work. The ttl may be negative, but the difference between now and the ttl would not be. Maybe we can check if ttl itself is negative. If that is so, we avoid checking the expiration at all.

Reply:
I see! you are right! Do we have some test case to cover this? Is it possible to add it?